src/terrain/KM_Terrain.pas

```diff
@@ -16,6 +16,27 @@ interface
                     vu_NESW);   //Vertex is used NE-SW like this: /
   TFenceType = (fncNone, fncCorn, fncWine, fncHousePlan, fncHouseFence);
 
+  TKMTileChangeType = (tctTerrain, tctHeight, tctObject);
+
+  TKMTileChangeTypeSet = set of TKMTileChangeType;
+
+  TKMTerrainTileBrief = packed record

```

Comment:
@Kromster80 suggested to me to use packed record, because I had some problems with memory - we also make a backup copy of all Land array and after it we invoke UpdateWalkConnect on all of the array (possible 255*255 Map). So that is why.
Probably we can remove packed now, sinse some optimisations was added afterwards.
@Kromster80 what do you think ?